lib/promscrape/discovery/ovhcloud/common.go

```diff
@@ -0,0 +1,105 @@
+package ovhcloud
+
+import (
+	"crypto/sha1"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/netip"
+	"strconv"
+	"time"
+)
+
+func getAuthHeaders(cfg *apiConfig, headers http.Header, endpoint, path string) (http.Header, error) {
+	headers = setGeneralHeaders(cfg, headers)
+
+	timeDelta, err := getTimeDelta(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	timestamp := time.Now().Add(-timeDelta).Unix()
+	headers.Set("X-Ovh-Timestamp", strconv.FormatInt(timestamp, 10))
+	headers.Add("X-Ovh-Consumer", cfg.consumerKey)
+
+	h := sha1.New()
+	h.Write([]byte(fmt.Sprintf("%s+%s+%s+%s+%s+%d",
+		cfg.applicationSecret,
+		cfg.consumerKey,
+		"GET",
+		endpoint+path,
+		"", // no body contained in any service discovery request, so it's set to empty by default
+		timestamp,
+	)))
+	headers.Set("X-Ovh-Signature", fmt.Sprintf("$1$%x", h.Sum(nil)))
+	return headers, nil
+}
+
+func setGeneralHeaders(cfg *apiConfig, headers http.Header) http.Header {
+	headers.Set("X-Ovh-Application", cfg.applicationKey)
+	headers.Set("Accept", "application/json")
+	headers.Set("User-Agent", "github.com/VictoriaMetrics/VictoriaMetrics")
+	return headers
+}
+
+func getServerTime(cfg *apiConfig) (*time.Time, error) {
+	resp, err := cfg.client.GetAPIResponseWithReqParams("/auth/time", func(req *http.Request) {
+		req.Header = setGeneralHeaders(cfg, req.Header)
+	})
+	if err != nil {
+		return nil, err
+	}
+	ts, err := strconv.ParseInt(string(resp), 10, 0)
+	if err != nil {
+		return nil, fmt.Errorf("parse ovh response to timestamp failed: %w", err)
+	}
+	serverTime := time.Unix(ts, 0)
+	return &serverTime, nil
+}
+
+func getTimeDelta(cfg *apiConfig) (time.Duration, error) {
```

Comment:
Would you mind adding a comment explaining that delta should be calculated only once?

Reply:
done